### PR TITLE
Reorganize Contract Directory Structure

### LINKED ISSUE #115: Reorganize contract directory structure
Base token contracts, their interfaces and presets are all thrown into the same folder today. We should think of a better way to organize files, with a focus on discoverability and ease of import paths declaration.

### PR DESCRIPTION
**Pull request by andrew-fleming**
Resolves #115. Reorganizes tests to match token contract directories.

### CODE REVIEW

**Comment by martriay** [comment]
Continued conversation on https://github.com/OpenZeppelin/cairo-contracts/issues/115#issuecomment-1017982532

File: openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
 from starkware.cairo.common.uint256 import Uint256
 
-from contracts.token.ERC721_base import (
+from openzeppelin.token.erc721.library.ERC721_base import (
```
Comment (martriay) [inline openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo:6]:
```suggestion
from openzeppelin.token.erc721.library import (
```

the idea is to go with `library` directly, no more references to "base contracts"

File: openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo [hunk 2]
```diff
@@ -21,16 +21,16 @@ from contracts.token.ERC721_base import (
     ERC721_setTokenURI
 )
 
-from contracts.ERC165_base import ERC165_supports_interface
+from openzeppelin.ERC165_base import ERC165_supports_interface
 
-from contracts.Pausable_base import (
+from openzeppelin.Pausable_base import (
```
Comment (martriay) [inline openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo:26]:
```suggestion
from openzeppelin.security.pausable import (
```
what do you think of putting pausable under `security` to mimic the [solidity lib](https://github.com/OpenZeppelin/openzeppelin-contracts/tree/master/contracts/security) categorization?

File: openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo [hunk 3]
```diff
@@ -21,16 +21,16 @@ from contracts.token.ERC721_base import (
...
     Pausable_paused,
     Pausable_pause,
     Pausable_unpause,
     Pausable_when_not_paused
 )
 
-from contracts.Ownable_base import (
+from openzeppelin.Ownable_base import (
```
Comment (martriay) [inline openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo:33]: we could follow [the solidity lib](https://github.com/OpenZeppelin/openzeppelin-contracts/tree/master/contracts/access) and move Ownable under `access/`
```suggestion
from openzeppelin.access.ownable import (
```

and we can move the current `Ownable.cairo` implementation to the testing folder where I think it belongs. I wouldn't consider it a preset.

File: openzeppelin/token/erc721/library/ERC721_Enumerable_base.cairo [hunk 4]
```diff
@@ -17,11 +17,9 @@ from contracts.token.ERC721_base import (
     ERC721_burn
 )
 
-from contracts.ERC165_base import (
-    ERC165_register_interface
-)
+from openzeppelin.ERC165_base import ERC165_register_interface
```
Comment (martriay) [inline openzeppelin/token/erc721/library/ERC721_Enumerable_base.cairo:20]:
```suggestion
from openzeppelin.introspection.ERC165 import ERC165_register_interface
```
165 can go under introspection

File: openzeppelin/token/erc721/library/ERC721_base.cairo [hunk 5]
```diff
@@ -6,18 +6,18 @@ from starkware.cairo.common.alloc import alloc
 from starkware.starknet.common.syscalls import get_caller_address
 from starkware.cairo.common.uint256 import Uint256, uint256_check
 
-from contracts.utils.safemath import (
+from openzeppelin.utils.safemath import (
```
Comment (martriay) [inline openzeppelin/token/erc721/library/ERC721_base.cairo:9]: i'm not sure `safemath` is just an util. maybe if we have a `security/` folder for `pausable.cairo`, it makes sense to have this one there too

**Review by martriay — CHANGES_REQUESTED** [review]
Looking very good! Left some comments

File: openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo (hunk 1 above)
Comment (andrew-fleming) [reply openzeppelin/token/erc721/ERC721_Mintable_Pausable.cairo:6]: Ahhhh I see. On it!

**Comment by andrew-fleming** [comment]
I left `Account` and `IAccount` as loose contracts without a directory for now. I figure this change should be in #185 and maybe have an `account` directory?

I left `AddressRegistry` as a loose contract as well, as I'm not sure what directory fits it best.

**Comment by andrew-fleming** [comment]
> I left `Account` and `IAccount` as loose contracts without a directory for now. I figure this change should be in #185 and maybe have an `account` directory?
> 
> I left `AddressRegistry` as a loose contract as well, as I'm not sure what directory fits it best.

This PR now includes the `account` directory with `Account`, `IAccount`, and `AddressRegistry`. If there's a better spot for `AddressRegistry`, we can certainly move it.

**Comment by martriay** [comment]
Looks good! Great work in here :)

`AddressRegistry` definitely belongs to Account, at least for now.

**Review by martriay — APPROVED** [review]
Looks good! Great work in here :)

AddressRegistry definitely belongs to Account, at least for now.

**Pull request merged**